drivers/wifi/eswifi/eswifi_socket_offload.c

```diff
@@ -0,0 +1,562 @@
+		pkt = k_fifo_get(&socket->fifo, K_NO_WAIT);
+		if (!pkt)
+			break;
+		net_pkt_unref(pkt);
+
+	}
+	if (--socket->usage <= 0) {
+		memset(socket, 0, sizeof(*socket));
+	}
+	eswifi_unlock(eswifi);
+
+	return 0;
+}
+
+static int eswifi_socket_open(int family, int type, int proto)
+{
+	struct eswifi_off_socket *socket = NULL;
+	int err, i;
+
+	LOG_DBG("");
+
+	if (family != AF_INET) {
+		LOG_ERR("Only AF_INET is supported!");
+		return -EPFNOSUPPORT;
+	}
+
+	eswifi_lock(eswifi);
+
+	/* pickup available socket */
+	for (i = 0; i < ESWIFI_OFFLOAD_MAX_SOCKETS; i++) {
+		if (!eswifi->socket[i].context) {
+			socket = &eswifi->socket[i];
+			socket->index = i;
+			/* Assign dummy context for socket(50CE) */
+			eswifi->socket[i].context = (struct net_context *)0x50CE;
+			break;
+		}
+	}
+
+	if (!socket) {
+		LOG_ERR("No socket resource available");
+		eswifi_unlock(eswifi);
+		return -ENOMEM;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS) &&
+	    proto >= IPPROTO_TLS_1_0 && proto <= IPPROTO_TLS_1_2) {
+		socket->type = ESWIFI_TRANSPORT_TCP_SSL;
+	} else if (proto == IPPROTO_TCP) {
+		socket->type = ESWIFI_TRANSPORT_TCP;
+	} else if (proto == IPPROTO_UDP) {
+		socket->type = ESWIFI_TRANSPORT_UDP;
+	} else {
+		LOG_ERR("Only TCP & UDP is supported");
+		eswifi_unlock(eswifi);
+		return -EPFNOSUPPORT;
+	}
+
+	err = __select_socket(eswifi, socket->index);
+	if (err < 0) {
+		LOG_ERR("Unable to select socket %u", socket->index);
+		eswifi_unlock(eswifi);
+		return -EIO;
+	}
+
+	snprintf(eswifi->buf, sizeof(eswifi->buf), "P1=%d\r", socket->type);
+
+	err = eswifi_at_cmd(eswifi, eswifi->buf);
+	if (err < 0) {
+		LOG_ERR("Unable to set transport protocol");
+		eswifi_unlock(eswifi);
+		return -EIO;
+	}
+
+	LOG_DBG("Socket index %d", socket->index);
+	k_fifo_init(&socket->fifo);
+	k_sem_init(&socket->read_sem, 0, 100);
+	k_delayed_work_init(&socket->read_work, eswifi_off_read_work);
+	socket->usage = 1;
+	k_delayed_work_submit_to_queue(&eswifi->work_q, &socket->read_work,
+					500);
+	socket->prev_pkt_rem = NULL;
+	eswifi_unlock(eswifi);
+
+	return 0;
+}
+
+static int eswifi_socket_poll(struct pollfd *fds, int nfds, int msecs)
+{
+	struct eswifi_off_socket *socket;
+	int sock, ret;
+
+	if (nfds > 1 ||
+	    fds[0].fd  > ESWIFI_OFFLOAD_MAX_SOCKETS)
+		return -EINVAL;
+
+	eswifi_lock(eswifi);
+	sock = fds[0].fd;
+	socket = &eswifi->socket[sock];
+	eswifi_unlock(eswifi);
+	if (socket->state != ESWIFI_SOCKET_STATE_CONNECTED)
+		return -EINVAL;
+
+	ret = k_sem_take(&socket->read_sem, msecs);
+	return ret;
+}
+
+struct socket_offload eswifi_socket_ops = {
+	/* POSIX Socket Functions: */
+	.socket = eswifi_socket_open,
+	.close = eswifi_socket_close,
+	.accept = eswifi_socket_accept,
+	.connect = eswifi_socket_connect,
+	.setsockopt = eswifi_socket_setsockopt,
+	.recv = eswifi_socket_recv,
+	.send = eswifi_socket_send,
+	.poll = eswifi_socket_poll,
+};
+
+int eswifi_socket_offload_init(struct eswifi_dev *leswifi)
+{
+#ifdef CONFIG_NET_SOCKETS_OFFLOAD
```

Comment:
Why? eswifi_socket_offload is not built anyway if CONFIG_NET_SOCKETS_OFFLOAD is not set

Reply:
It was from draft version, I will remove it